packages/slice-machine/tests/src/modules/customTypes/index.test.ts

```diff
@@ -101,6 +102,33 @@ describe("[Custom types module]", () => {
         .next()
         .put(modalCloseCreator({ modalKey: ModalKeysEnum.CREATE_CUSTOM_TYPE }));
       saga.next().put(push("/cts/id"));
+      saga.next().put(
+        openToasterCreator({
+          message: "Custom type saved",
+          type: ToasterType.SUCCESS,
+        })
+      );
+      saga.next().isDone();
+    });
+    it("should call the api and dispatch the good actions on failure", () => {
+      const actionPayload = { id: "id", label: "label", repeatable: true };
+      const customTypeCreated = createCustomType(
+        actionPayload.id,
+        actionPayload.label,
+        actionPayload.repeatable
+      );
+      const saga = testSaga(
+        createCustomTypeSaga,
+        createCustomTypeCreator.request(actionPayload)
+      );
+
+      saga.next().call(saveCustomType, customTypeCreated, {});
+      saga.throw(new Error()).put(
```

Comment:
not sure to understand those tests because I don't see any expect.

Reply:
Yeah this test use a lib redux-saga-test-plan, a lib that is very usefull for unit testing generator function, each time I call next() the generator is jumping in the next step. Then the lib provide helpers 'call', 'throw', 'put' and others to expect specific side effects from the saga